### ISSUE TITLE
E3002 false positive when using Fn::If in AWS::Transfer::User  SshPublicKeys

### ISSUE THREAD

**Issue by agostain** [aws-cloudformation/cfn-lint#2156]
*cfn-lint version: cfn-lint 0.54.3*

E3002 false positive when using Fn::If in AWS::Transfer::User  SshPublicKeys

Please provide as much information as possible:
* Template linting issues:
  * Please provide a CloudFormation sample that generated the issue.

```
AWSTemplateFormatVersion: '2010-09-09'
Description: Transfer Family User Creation Stack

Parameters:
  PartnerName:
    Description: "Partner Application"
    Type: String
  Direction:
    Description: "Direction"
    Type: String
    AllowedValues:
      - "get"
      - "put"
  SshPublicKey:
    Description: "Ssh public key for the user"
    Type: List<String>

# ....

Conditions:
  UseGet: !Equals [!Ref Direction, get]
  HaveSshPublicKey: !Not [!Equals [!Join ['', !Ref SshPublicKey], '']]

Resources:
  TFUser:
    Type: AWS::Transfer::User
    Properties:
      HomeDirectoryMappings: 
        - Entry: /
          Target: !Join
                    - ""
                    - - "/"
                      - !Ref FileSystemId
                      - !If [ UseGet, !Sub "/dev/xxx/out/${PartnerName}", !Sub "/dev/xxx/in/${PartnerName}"]
      HomeDirectoryType: "LOGICAL"
      PosixProfile:
        Gid: 1505 
        Uid: 1505
      Role: !Ref UserRoleArn        # Required
      ServerId: !Ref TFServerId  # Required
      SshPublicKeys: !If
          - HaveSshPublicKey
          - !Ref SshPublicKey
          - !Ref "AWS::NoValue"
      UserName: "testuser"
```

  * If present, please add links to the (official) documentation for clarification.
  * (https://docs.aws.amazon.com/de_de/AWSCloudFormation/latest/UserGuide/aws-resource-transfer-user.html#cfn-transfer-user-sshpublickeys)
  * The above CloudFormation is valid, however, cfn-lint gripes with E3002 when using a conditional statement
  Bug report: E3002 Property SshPublicKeys should be of type List for resource TFUser
  
  * Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files
  * latest version used - cfn-lint 0.54.3

 
  

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)
